src/main/java/org/jenkinsci/plugins/workflow/cps/steps/ingroovy/GroovyCompiler.java

```diff
@@ -0,0 +1,106 @@
+package org.jenkinsci.plugins.workflow.cps.steps.ingroovy;
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.Iterators;
+import groovy.lang.Binding;
+import groovy.lang.GroovyShell;
+import hudson.Extension;
+import hudson.PluginWrapper;
+import jenkins.model.Jenkins;
+import org.codehaus.groovy.control.CompilerConfiguration;
+import org.jenkinsci.plugins.workflow.cps.steps.ingroovy.StepInGroovy.StepDescriptorInGroovy;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Used to compile Groovy step implementation to be able to reflect on its properties.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+@Extension
+@Restricted(NoExternalUse.class) // SezPoz demands that this is public, but this is an implementation detail in this plugin
+public class GroovyCompiler {
```

Comment:
I was gonna throw a note on this because I was sure there was already a class in Groovy named this, but no! There isn't! So moving on... =)